cmd/podman/port.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	"github.com/projectatomic/libpod/libpod"
+	"strings"
+	"strconv"
+)
+
+var (
+	portFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name: "all, a",
+			Usage: "display port information for all containers",
+		},
+		LatestFlag,
+	}
+	portDescription = `
+   podman port
+	
+	List port mappings for the CONTAINER, or lookup the public-facing port that is NAT-ed to the PRIVATE_PORT
+`
+
+	portCommand = cli.Command{
+		Name:        "port",
+		Usage:       "List port mappings or a specific mapping for the container",
+		Description: portDescription,
+		Flags:       portFlags,
+		Action:      portCmd,
+		ArgsUsage:   "CONTAINER-NAME [mapping]",
+	}
+)
+
+func portCmd(c *cli.Context) error {
+	var userProto, containerName string
+	var userPort int
+	var container *libpod.Container
+	var containers []*libpod.Container
+
+	args := c.Args()
+	if err := validateFlags(c, portFlags); err != nil {
+		return err
+	}
+
+	if c.Bool("latest") && c.Bool("all") {
+		return errors.Errorf("the 'all' and 'latest' options cannot be used together")
+	}
+	if c.Bool("all") && len(args) > 0 {
+		return errors.Errorf("no additional arguments can be used with 'all'")
+	}
+	if len(args) == 0 && !c.Bool("latest") && !c.Bool("all") {
+		return errors.Errorf("you must supply a running container name or id")
+	}
+	if !c.Bool("latest") && !c.Bool("all") {
+		containerName = args[0]
+	}
+
+	port := ""
+	if len(args) > 1 && !c.Bool("latest") {
+		port = args[1]
+	}
+	if len(args) == 1 && c.Bool("latest"){
+		port = args[0]
+	}
+	if port != "" {
```

Comment:
Can I use port with --all?  If yes then arg[0]would be the port?  Should we just make the port mapping be --port and go away from the Docker default?

Reply:
Cannot use port with --all.  

I was trying to keep compatibility but the parsing would be way easier if the port and protocol were called out with --options.